internal/component/prometheus/exporter/github/github.go

```diff
@@ -21,8 +21,9 @@ func init() {
 	})
 }
 
-func createExporter(opts component.Options, args component.Arguments, defaultInstanceKey string) (integrations.Integration, string, error) {
+func createExporter(opts component.Options, args component.Arguments) (integrations.Integration, string, error) {
 	a := args.(Arguments)
+	defaultInstanceKey := opts.ID // if cannot resolve instance key, use the component ID
```

Comment:
What is the reason we don't do `WarningIfUsedInCluster` here? Is it to avoid spamming the logs?